src/wdio/services/TerraCommands/visual-regression.js

```diff
@@ -0,0 +1,140 @@
+
+  Object.entries(styleProperties).forEach(([key, value]) => {
+    global.it(`themed [${key}]`, () => {
+      global.browser.execute(`document.documentElement.style.setProperty('${key}', '${value}')`);
+      global.expect(global.browser.checkElement(selector)).to.matchReference();
+    });
+  });
+};
+
+/** Helper method to create a useful test descripton.
+  * @property {String} matchType - Specifies the type of matchReference assertion. Either 'withinTolerance'
+  *   or 'exactly'.
+  */
+const getTestDescription = matchType => (
+  matchType === 'withinTolerance' ? 'be within the mismatch tolerance' : 'match screenshot exactly'
+);
+
+/**
+* Determines the screenshot options and test names needed to call the `itMatchesScreenshot`
+* mocha-chai method.
+* @property {Array} testArguments - The test arguments passed to the `matchScreenshotWithinTolerance`
+*    or `matchScreenshotExactly` methods.
+* @property {String} matchType - Specifies the type of matchReference assertion. Either 'withinTolerance'
+*    or 'exactly'.
+*/
+const matchScreenshot = (testArguments, matchType) => {
+  const { name, selector, viewports, options } = determineScreenshotOptions(...testArguments);
+
+  const testDescription = getTestDescription(matchType);
+
+  if (viewports.length) {
+    // Create an assertion for each viewport such that better information is provided if failure occurs for a screenshot
+    viewports.forEach((viewport) => {
+      const testName = `[${name}] to ${testDescription} for ${viewport.name} viewport`;
+      options.viewports = [viewport];
+      itMatchesScreenshot(testName, selector, options, matchType);
+    });
+  } else {
+    itMatchesScreenshot(`[${name}] to ${testDescription}`, selector, options, matchType);
+  }
+};
+
+/**
+* Mocha-chai wrapper method to capture screenshots of a specified element and assert the
+* screenshot comparision results are within the mismatch tolerance.
+* @property {Array} args - The list of test arguments to parse. Accepted Arguments:
+*    - String (optional): the test case name. Default name is 'default'
+*    - Object (optional): the test options. Options include selector, viewports,
+*        misMatchTolerance and viewportChangePause.
+*    Note: args list order should be: name, then options when using both.
+*/
+const matchScreenshotWithinTolerance = (...args) => {
+  matchScreenshot(args, 'withinTolerance');
+};
+
+/**
+* Mocha-chai wrapper method to capture screenshots of a specified element and assert the
+* screenshot comparision results are an exact match.
+* @property {Array} args - The list of test arguments to parse. Accepted Arguments:
+*    - String (optional): the test case name. Default name is 'default'
+*    - Object (optional): the test options. Options include selector, viewports,
+*        misMatchTolerance and viewportChangePause.
+*     Note: args list order should be: name, then options when using both.
+*/
+const matchScreenshotExactly = (...args) => {
```

Comment:
This method does scare me. We have had issues with being too pixel perfect with projects I've worked on for other teams, and it makes me a little concerned this method could be used way too much. Should we have some kind of documentation saying this should rarely be used?

Reply:
I'm not sure to be honest. @mjhenkes thoughts?

From a standalone component standpoint, it makes sense to want a closer to perfect image, but IMO from an application standpoint consumer terra-components, it would be best to use `matchScreenshot()` method given we might a small style change that they wouldn't need to be concerned with.

All of our test will be using `matchScreenshot()`, which I imagine most people will be referencing when starting to write tests.